FO:  Remove Front controller deprecated properties

nenes25 opened the pull request:
<!-----------------------------------------------------------------------------
Thank you for contributing to the PrestaShop project! 

Please take the time to edit the "Answers" rows below with the necessary information.

Check out our contribution guidelines to find out how to complete it:
https://devdocs.prestashop.com/1.7/contribute/contribution-guidelines/#pull-requests
------------------------------------------------------------------------------>

| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop 
| Description?      | Remove deprecated class attributes and global variables from front controller (except $useSSL )
| Type?             |  improvement
| Category?         | FO
| BC breaks?        | Old modules which relies on usages of the global variables : $cookie, $smarty, $cart, $iso, $defaultCountry, $protocol_link, $protocol_content, $link, $css_files, $js_files, $currency should change to use variables from the context instead.
| Deprecations?     | yes / no
| Fixed ticket?     | Relative to  #26293
| How to test?      |  Global tests should not fail
| Possible impacts? | Old modules relying on globals will not works anymore ( but usage is discouraged since the release of prestashop 1.5.0.1 ... ) As discussed in this PR the useSSL var can have impacts and will be treated in another PR.


<!-- Click the form's "Preview" button to make sure the table is functional in GitHub. Thank you! -->

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/27756)
<!-- Reviewable:end -->

PrestaEdit reviewed (changes requested):
Regarding the remove of the $useSSL var into the global, you will need to assign is or make a test isset into the line 290 cause it's check the value of it.

Progi1984 reviewed (commented):
@nenes25 Thank you for your contribution. Could you check the @PrestaEdit feedback ?

nenes25 commented:
Hello @PrestaEdit and @Progi1984,

Thanks for your feedback, i've updated and removed all references to the global $useSSL.

Initially i don't think that this PR will have a lot of impacts
But it seems that the remove of $useSSL can have significant ones.
Is it a better idea to split ? ( One PR for $useSSL and one for the others ? ) 

Regards,

matks commented:
> Hello @PrestaEdit and @Progi1984,
> 
> Thanks for your feedback, i've updated and removed all references to the global $useSSL.
> 
> Initially i don't think that this PR will have a lot of impacts But it seems that the remove of $useSSL can have significant ones. Is it a better idea to split ? ( One PR for $useSSL and one for the others ? )
> 
> Regards,

I think the split will be better because you will have 1 PR very easy to review and test and the other one, that concerns `$useSSL` that might require deep review and deep QA. So 2 different PRs -> better split 😄

matks reviewed (dismissed):
Also our Fixer of codestyle, PHP CS Fixer, mentions a little codestyle issue

nenes25 committed:
Remove deprecated class attributes and global variables from front controller ( Keep $useSSL )

nenes25 commented:
OK it's done :) , thanks all for the feedbacks ! :smile: 
I'll remove the $useSSL in another PR.
Let's wait to see if this one can be merged quickly.

Regards,

Progi1984 reviewed (changes requested):
@nenes25 Thank you for your contribution. The code is correct (thanks for it). Could you document BC breaks, please ?

nenes25 commented:
@Progi1984 the PR description is updated with BC break 
Let me know if it's enough  :smile:

Progi1984 commented:
@PrestaShop/qa-functional Do you think we need to test it ?

Robin-Fischer-PS commented:
@Progi1984 We should just pass the automated test campaign on this one, and it will be OK.

HanaRebaiQA reviewed (approved):
Hello @nenes25 

I have checked this PR.
All automated tests are passed :heavy_check_mark: 

So, it's QA :heavy_check_mark: 

Thanks!
